AsyncDisplayKit/Layout/ASLayoutSpec.mm

```diff
@@ -168,11 +169,13 @@ - (NSArray *)children
 
 - (ASEnvironmentState)environmentState
 {
+  ASDN::MutexLocker l(_propertyLock);
```

Comment:
Agree we have to be careful with adding locks in ASEnvironment. E.g. we didn't lock layout options in the past before ASEnvironment and as we added locks for the ASEnvironmentLayoutOptions deadlocks happened. Especially if we traverse the node tree's from bottom up (e.g. it happens if called `setNeedsLayout`) and the same time the layout happens from top down (e.g. call `measureWithSizeRange:` directly)